R/read-deprecated.R

```diff
@@ -21,243 +31,134 @@
-      details = "Please use the `...` argument instead to filter data."
-    )
-  }
-
-  # Error message if multiple criteria selected
-  if (lifecycle::is_present(chrom) && lifecycle::is_present(gene)) {
-    abort(c(
-      "Multiple filtering criteria selected.",
-      x = "Cannot filter on both `chrom` and `gene`.",
-      i = "Select only one piece of information to filter on."
-    ))
-  }
-
-  # Error if any file is empty
-  if (purrr::some(list(.ref_file, .alt_file, .cov_file), empty_file)) {
-    empty <- purrr::detect(list(.ref_file, .alt_file, .cov_file), empty_file)
-    abort(c(
-      "Unable to read files.",
-      x = glue('"{empty}" is an empty file.')
-    ))
-  }
-
-  # Read in the three files
-  if (lifecycle::is_present(chrom) || lifecycle::is_present(gene)) {
-    tables <- purrr::pmap(
-      list(
-        file = c(.ref_file, .alt_file, .cov_file),
-        name = c("ref_umi_count", "alt_umi_count", "coverage")
-      ),
-      deprec_read_file,
-      chrom = chrom,
-      gene = gene
-    )
-  } else {
-    tables <- purrr::pmap(
-      list(
-        .file = c(.ref_file, .alt_file, .cov_file),
-        .name = c("ref_umi_count", "alt_umi_count", "coverage")
-      ),
-      read_file,
-      ...
-    )
-  }
-
-  # Determine overlapping columns
-  by <- purrr::reduce(purrr::map(tables, colnames), intersect)
+  lifecycle::deprecate_warn(
+    when = "0.2.0",
+    what = "read()",
+    with = "read_tbl_ref_alt_cov()"
+  )
 
-  # Combine three tibbles together
-  purrr::reduce(tables, dplyr::full_join, by = by)
+  read_tbl_ref_alt_cov(
+    .tbl_ref = .ref_file,
+    .tbl_alt = .alt_file,
+    .tbl_cov = .cov_file,
+    ...,
+    chrom = chrom,
+    gene = gene
+  )
 }
 
-#' @rdname read
+#' @rdname read-deprecated
 #' @export
 read_file <- function(.file, ..., .name = "value") {
-  dots <- enquos(..., .ignore_empty = "all")
-  check_named(dots)
-
-  if (empty_file(.file)) {
-    return(tibble::tibble())
-  }
-
-  # Read in complete header
-  header <- .file %>%
-    vroom::vroom(col_names = FALSE, show_col_types = FALSE, n_max = 6) %>%
-    tibble::rownames_to_column() %>%
-    tidyr::pivot_longer(-.data$rowname) %>%
-    tidyr::pivot_wider(
-      names_from = .data$rowname,
-      values_from = .data$value
-    ) %>%
-    janitor::row_to_names(1) %>%
-    janitor::clean_names()
-
-  # Filter the header based on conditions specified
-  tryCatch(
-    filter_header <- dplyr::filter(header, ...),
-    error = function(e) {
-      e <- rlang::catch_cnd(dplyr::filter(header, ...))
-      msg <- e$message %>%
-        stringr::str_replace("filter", "read") %>%
-        stringr::str_replace("object", "Object") %>%
-        stringr::str_c(".")
-      objects <- stringr::str_c("'", colnames(header)[-1], "'")
-      abort(c(
-        msg,
-        i = cli::pluralize("Available objects are {objects}.")
-      ))
-    }
+  msg <- paste(
+    "The function has been replaced by three more specific functions:\n",
```

Comment:
The function _____ has been replaced...

Reply:
`lifecycle::deprecate_warn()` actually prints the function name in the first line of the warning, so this is fine as is.